gossip/sfc_index.go

```diff
@@ -201,20 +201,55 @@ func (s *Service) processSfc(block *inter.Block, receipts types.Receipts, blockF
 			}
 
 			// Deactivate delegators
-			if l.Topics[0] == sfcpos.Topics.PreparedToWithdrawDelegation && len(l.Topics) > 1 {
+			if (l.Topics[0] == sfcpos.Topics.DeactivatedDelegation || l.Topics[0] == sfcpos.Topics.PreparedToWithdrawDelegation) && len(l.Topics) > 1 {
```

Comment:
It's more logical to check for "len(l.Topics) > 1"  before any other condition like "l.Topics[0] == .... "

Reply:
There cannot be less than 1 topic